### PR TITLE
Add gpg/ed25519 support

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
None

**Description of the changes being introduced by the pull request**:
- Add `securesystemslib.gpg.eddsa` to implement the algorithmic-specific parts to parse
and verify EdDSA/ed25519 public keys and signatures generated with gpg, in the same fashion as the `rsa` and `dsa` modules.

- Update gpg-specific formats, so that gpg interface functions (create_signature, export_pubkey, verify_signature) accept ed25519 keys.

- Add Travis-compatible (gpg v2.1.11) keychain with one ed25519 key and basic tests for the gpg interface functions.

- *Fix a few comments/docstrings on the go*

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by lukpueh** [comment]
Big kudos to @SantiagoTorres for excavating a relevant detail from RFC8032!

File: securesystemslib/gpg/eddsa.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+"""
+<Module Name>
+  dsa.py
```
Comment (JustinCappos) [inline securesystemslib/gpg/eddsa.py]: doesn't match the file name

**Comment by awwad** [comment]
Seems to function as expected.  👍
I added two minor comments -- one is a correction re. a misunderstanding of RFC 4880, and the other is possibly an extraneous comment.

I'm going to toy with it a bit more and possibly add some more testing.

File: securesystemslib/gpg/common.py [hunk 2]
```diff
@@ -647,7 +650,7 @@ def parse_signature_packet(data, supported_signature_types=None,
   if signature_algorithm not in SUPPORTED_SIGNATURE_ALGORITHMS:
     raise ValueError("Signature algorithm '{}' not "
         "supported, please verify that your gpg configuration is creating "
-        "either DSA or RSA signatures (see RFC4880 9.1. Public-Key "
+        "either DSA, RSA, or EdDSA signatures (see RFC4880 9.1. Public-Key "
```
Comment (lukpueh) [inline securesystemslib/gpg/common.py]: Thanks for spotting this.  There's a similar error message that needs updating in `parse_pubkey_payload`.

File: securesystemslib/gpg/common.py [hunk 3]
```diff
@@ -632,7 +632,10 @@ def parse_signature_packet(data, supported_signature_types=None,
     raise ValueError("Signature version '{}' not supported, must be one of "
         "{}.".format(version_number, SUPPORTED_SIGNATURE_PACKET_VERSIONS))
 
-  # here, we want to make sure the signature type is indeed PKCSV1.5 with RSA
+  # Here, we want to make sure the signature type is indeed a "signature of a
+  # binary document", type 0x00 from RFC 4880 section 5.2.1.  This is as
+  # opposed to a revocation signature, or a signature of one key over another,
+  # or a persona certification, etc.
```
Comment (lukpueh) [inline securesystemslib/gpg/common.py]: `SIGNATURE_TYPE_BINARY` is only the default signature type that's checked for in `parse_signature_packet`. But you can pass other `supported_signature_types` as we e.g. do in `_assign_certified_key_info` (`SIGNATURE_TYPE_CERTIFICATES`) and `_get_verified_subkeys` (`SIGNATURE_TYPE_SUB_KEY_BINDING`).

**Comment by awwad** [comment]
Can we merge this?  It's working well for me.  I'll provide some upstream content later.

**Pull request merged**